jupyterhub/templates/image-puller/job.yaml

```diff
@@ -9,24 +9,24 @@ command.
 apiVersion: batch/v1
 kind: Job
 metadata:
-  name: hook-image-awaiter-{{ .Release.Time.Seconds }}
+  name: hook-image-awaiter
   labels:
     {{- include "jupyterhub.labels" . | nindent 4 }}
     hub.jupyter.org/deletable: "true"
   annotations:
     "helm.sh/hook": pre-install,pre-upgrade
-    "helm.sh/hook-delete-policy": hook-succeeded,hook-failed
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
```

Comment:
What happens on hook-failed?

Reply:
great point, this bothinh before and nothing when it is removed, this was a bug id say but helm dis not consider it very important.

only the last created Job resource the among hook resources will be deleted if this deletion policy is around on a Job resource. For a daemon set it does kot matter at all and with ko warning if another job etc failed in the dame helm upgrade etc.

removing it as it is dead code causing confusion  but i should perhaps add notes about this

Reply:
I edited the reply above, it was not really readable before - will stop writing from mobile without reviewing my text haha sorry @minrk!